src/EVM/Expr.hs

```diff
@@ -1292,8 +1298,13 @@ simplifyProp prop =
 flattenProps :: [Prop] -> [Prop]
 flattenProps [] = []
 flattenProps (a:ax) = case a of
-  PAnd x1 x2 -> x1:x2:flattenProps ax
+  PAnd x1 x2 -> (subFlatten x1) ++ (subFlatten x2) ++ flattenProps ax
   x -> x:flattenProps ax
+  where
+    subFlatten :: Prop -> [Prop]
+    subFlatten b = case b of
+      PAnd y1 y2 -> flattenProps [y1,y2]
+      _ -> [b]
```

Comment:
Can we do without the helper function? Like this?
```haskell
flattenProps :: [Prop] -> [Prop]
flattenProps [] = []
flattenProps (a:ax) = case a of
  PAnd x1 x2 -> flattenProps [x1] ++ flattenProps [x2] ++ flattenProps ax
  x -> x:flattenProps ax
```

Would the helper function be more efficient for any reason?

Reply:
Oh, of course! Good point :D Fixed!